### PR TITLE
Added QuickSort in Bash

### LINKED ISSUE #1228: Add Quick Sort in Bash
See requirements: https://sample-programs.therenegadecoder.com/projects/quick-sort/
Add snipper here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/archive/b/bash

### PR DESCRIPTION
**Pull request by tehtbl**
- [x] I fixed #1228 

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [x] I added solution citations when necessary (see [plagiarism][2])

### CODE REVIEW

File: archive/b/bash/quicksort.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
```
Comment (jrg94) [inline archive/b/bash/quicksort.sh]: I'm getting bizarre errors here when I run this in an online editor:

```
jdoodle.sh: line 2: $'\r': command not found
jdoodle.sh: line 3: syntax error near unexpected token `$'{\r''
jdoodle.sh: line 3: `qsort_iter() {
'
```

Comment (tehtbl) [reply archive/b/bash/quicksort.sh]: Could you please tell me at which online editor you've tried this out?

When I am using the script locally, everything works fine:
```
$ ./quicksort.sh 4 2 3 4 1 1 2 3 4
1 1 2 2 3 3 4 4 4
```

Comment (jrg94) [reply archive/b/bash/quicksort.sh]: I tried JDoodle and a couple others.

**Comment by tehtbl** [comment]
Hi @jrg94 

I've accidentally already deleted the fork on my side. So, I cannot resolve the conflict without write access to the master branch. So, could you please make the small changes to the README.md or should I close this PR and create another one?

cheers,
Thomas

**Comment by jrg94** [comment]
Yeah, let's go the refork route. I would fix the conflict myself, but it won't let me.

**Pull request closed without merging**